flux-fhir-analysis/src/lib.rs

```diff
@@ -142,6 +143,21 @@ fn item_bounds(
     Ok(rty::EarlyBinder(conv::conv_opaque_ty(genv, local_id.to_def_id(), opaque_ty, &wfckresults)?))
 }
 
+fn refparams_of(genv: &GlobalEnv, local_id: LocalDefId) -> QueryResult<rty::RefParams> {
```

Comment:
I still think we don't need this and we can put it as an extra field in `rty::Generics`. We would avoid the extra query and type.

To do that we just need to pass a `&[fhir::RefineParam]` (the result of `genv.map().get_refine_params()`) to `conv::conv_generics`. Everywhere else we create a `rty::Generics` we just fill the extra field with an empty list.

For easy access we could still have a method in `GlobalEnv`

```rust
fn refine_params_of(self: &GlobalEnv, def_id: DefId) -> QueryResult<List<RefineParam>> {
    self.generics_of(def_id).refine_params.clone()
}
```

Reply:
makes sense, will do!